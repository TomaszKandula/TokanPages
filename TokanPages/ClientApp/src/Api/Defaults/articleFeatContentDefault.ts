import { IArticleFeatContentDto } from "../Models"

export const articleFeatContentDefault: IArticleFeatContentDto =
{
    content: 
    {
        title: "",
        desc: "",
        text1: "",
        text2: "",
        button: "",
        image1: "",
        image2: "",
        image3: "",
        image4: ""
    }
}
