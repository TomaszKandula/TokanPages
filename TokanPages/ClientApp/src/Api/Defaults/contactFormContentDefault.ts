import { IContactFormContentDto } from "../Models"

export const contactFormContentDefault: IContactFormContentDto =
{
    content:
    {
        caption: "",
        text: "",
        button: ""
    }    
}
