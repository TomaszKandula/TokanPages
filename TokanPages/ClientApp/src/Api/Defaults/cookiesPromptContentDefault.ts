import { ICookiesPromptContentDto } from "../Models"

export const cookiesPromptContentDefault: ICookiesPromptContentDto =
{
    content: 
    {
        caption: "",
        text: "",
        button: "",
        days: 0
    }
}
