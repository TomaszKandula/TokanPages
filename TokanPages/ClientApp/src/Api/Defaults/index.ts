export { navigationDefault } from "./navigationDefault";
export { articleFeatDefault} from "./articleFeatDefault";
export { contactFormDefault } from "./contactFormDefault";
export { cookiesPromptDefault } from "./cookiesPromptDefault";
export { featuredDefault } from "./featuredDefault";
export { featuresDefault } from "./featuresDefault";
export { newsletterDefault } from "./newsletterDefault";
export { resetFormDefault } from "./resetFormDefault";
export { signinFormDefault } from "./signinFormDefault";
export { signupFormDefault } from "./signupFormDefault";
export { testimonialsDefault } from "./testimonialsDefault";
export { unsubscribeDefault } from "./unsubscribeDefault";
export { updateSubscriberDefault } from "./updateSubscriberDefault";
