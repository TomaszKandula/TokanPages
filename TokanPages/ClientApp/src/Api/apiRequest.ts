import axios, { Method } from "axios";
import { UNEXPECTED_ERROR } from "../Shared/constants";
import { GetUnexpectedStatusCode } from "../Shared/Modals/messageHelper";
import { IError } from "./Models";

export interface IRequest
{
    method: Method;
    url: string;
    data?: any;
}

export interface IResult
{
    isSucceeded: boolean;
    error: IError;
}

export default async function ApiRequest(request: IRequest): Promise<IResult>
{
    let results: IResult = 
    { 
        isSucceeded: false, 
        error: 
        { 
            ErrorCode: "UNEXPECTED_ERROR", 
            ErrorMessage: UNEXPECTED_ERROR, 
            ValidationErrors: [] 
        } 
    };

    await axios(
    { 
        method: request.method, 
        url: request.url, 
        data: request.data
    })
    .then(function (response) 
    {
        if (response.status === 200) 
        {
            results = 
            {
                isSucceeded: true, 
                error:         
                { 
                    ErrorCode: "", 
                    ErrorMessage: "", 
                    ValidationErrors: [] 
                } 
            };
        }
        else
        {
            results = 
            {
                isSucceeded: false, 
                error:  
                {
                    ErrorCode: "UNEXPECTED_STATUS", 
                    ErrorMessage: GetUnexpectedStatusCode(response.status), 
                    ValidationErrors: [] 
                }
            };
        }
    })
    .catch(function (error) 
    {
        console.error(error);
        results = 
        {
            isSucceeded: false, 
            error:  
            {
                ErrorCode: error.response.data.ErrorCode, 
                ErrorMessage: error.response.data.ErrorMessage, 
                ValidationErrors: error.response.data.ValidationErrors
            }
        };
    });
    
    return results;
}
