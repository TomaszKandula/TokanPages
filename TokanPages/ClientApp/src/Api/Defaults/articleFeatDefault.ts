import { IArticleFeat } from "Api/Models"

const articleFeatInitValues: IArticleFeat =
{
    content: 
    {
        title: "",
        desc: "",
        text1: "",
        text2: "",
        button: "",
        image1: "",
        image2: "",
        image3: "",
        image4: ""
    }
}

export { articleFeatInitValues }
