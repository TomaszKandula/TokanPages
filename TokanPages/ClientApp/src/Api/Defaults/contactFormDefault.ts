import { IContactForm } from "Api/Models"

export const contactFormInitValues: IContactForm =
{
    content:
    {
        caption: "",
        text: "",
        button: ""
    }    
}
