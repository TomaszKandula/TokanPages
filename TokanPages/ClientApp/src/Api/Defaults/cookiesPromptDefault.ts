import { ICookiesPrompt } from "Api/Models"

export const cookiesPromptDefault: ICookiesPrompt =
{
    content: 
    {
        caption: "",
        text: "",
        button: "",
        days: 0
    }
}
