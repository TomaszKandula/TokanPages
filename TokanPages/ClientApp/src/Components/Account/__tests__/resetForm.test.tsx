import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import ResetForm from "../resetForm";
import { resetFormDefault } from "../../../Api/Defaults";

describe("Test account component: ResetForm.", () => 
{
    it("Renders correctly '<ResetForm />' when content is loaded.", () => 
    {
        const tree = shallow(<ResetForm resetForm={resetFormDefault} isLoading={false} />);
        expect(tree).toMatchSnapshot();
    });
});
