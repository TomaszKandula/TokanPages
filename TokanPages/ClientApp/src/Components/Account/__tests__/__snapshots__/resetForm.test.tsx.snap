// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test account component: ResetForm. Renders correctly '<ResetForm />' when content is loaded. 1`] = `
<section>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Styled(MuiBox)
      pb={10}
      pt={18}
    >
      <WithStyles(ForwardRef(Card))
        elevation={4}
      >
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-card-1"
        >
          <Styled(MuiBox)
            mb={3}
            textAlign="center"
          >
            <AccountCircleIcon
              color="primary"
              style={
                Object {
                  "fontSize": 72,
                }
              }
            />
            <WithStyles(ForwardRef(Typography))
              color="textSecondary"
              component="h2"
              variant="h5"
            />
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={2}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="email"
                  fullWidth={true}
                  id="email"
                  label="Email address"
                  name="email"
                  required={true}
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
            <Styled(MuiBox)
              my={2}
            >
              <WithStyles(ForwardRef(Button))
                color="primary"
                fullWidth={true}
                variant="contained"
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
